### PR TITLE
#2401 updates is_accepted

### LINKED ISSUE #2401: DOI resolution falsely flagged as failed during prepub
**Describe the bug**
When an article is in prepub, the doi checker always flags the DOI resolution as failed

**Janeway version**
v1.3.10

**Additional context**
The article is wrapped with the decorator `article_accepted_or_later_required` which in turn calls `Article.is_accepted`. The latter has a list of hard-coded article stages in which prepublication is not listed.

### PR DESCRIPTION
**Pull request by ajrbyers**
Closes #2401

### CODE REVIEW

File: src/submission/models.py [hunk 1]
```diff
@@ -768,10 +773,20 @@ def get_pubid(self):
...
-        return self.stage == "Published" or self.stage == "Accepted" or self.stage == "Editor Copyediting"\
-            or self.stage == "Author Copyediting" or self.stage == "Final Copyediting"\
-            or self.stage == "Typesetting" or self.stage == "Proofing"
+        from core import models as core_models
+        if self.date_published:
+            return True
+
+        if ArticleStageLog.objects.filter(
+            article=self,
+            stage_to='Accepted',
```
Comment (mauromsl) [inline src/submission/models.py:782]: use `STAGE_ACCPETED` instead?

File: src/submission/models.py [hunk 2]
```diff
@@ -768,10 +773,20 @@ def get_pubid(self):
...
+        ).exists():
+            return True
+
+        if not self.journal.element_in_workflow('review') and self.stage not in NEW_ARTICLE_STAGES:
+            return True
```
Comment (mauromsl) [inline src/submission/models.py:787]: I hadn't thought of considering a  `NEW_ARTICLE_STAGES`, maybe now we can simplify the whole method to:

`self.stage not in (NEW_ARTICLE_STAGES + REVIEW_STAGES)`?

**Review by mauromsl — CHANGES_REQUESTED** [review]
One change request and one comment

File: src/submission/models.py (hunk 2 above)
Comment (ajrbyers) [reply src/submission/models.py:787]: NOICE.

**Comment by ajrbyers** [comment]
test this please

**Review by mauromsl — APPROVED** [review]

**Pull request merged**